observation/store_test.go

```diff
@@ -39,7 +39,7 @@ func TestStore_SaveAll(t *testing.T) {
 		db := graph.Test(true, true, true)
 
 		errorReporterMock := reportertest.NewImportErrorReporterMock(nil)
-		s := store.NewStore(idCache, db, errorReporterMock)
+		s := store.NewStore(idCache, db, errorReporterMock, true)
 
 		Convey("When save all is called", func() {
 
```

Comment:
Maybe we want to add an extra convey block, testing the case when the flag is false, and validating that GetNodeIDs is not called?

Reply:
Actually, we are not checking dependencies calls in the current tests, so it would be some extra work for both cases, and validate the calls against idCache.GetNodeIDs (maybe it would be a good idea to create a mock with `moq` for the DimensionIDCache interface, and then make observationtest.DimensionIDCache wrap it?) - if you think it's out of the scope of this task, feel free to disagree.

Reply:
I think all of that is beyond the scope of this task.